pkg/uploader/chart_uploader.go

```diff
@@ -29,7 +29,7 @@ type ChartUploader struct {
 	// Out is the location to write warning and info messages.
 	Out io.Writer
 	// Pusher collection for the operation
-	Pushers pusher.Providers
+	Pushers pusher.ProvidersResolver
```

Comment:
Can't you add your test provider to Providers to avoid the interface change?

Reply:
I don't think I understand this comment. Provider is not an interface, it's a struct. I can't mock out a function on a struct, so that's why I have an interface where Provider is one implementation of that interface.